linkerd/http-route/src/http/filter/modify_request_header.rs

```diff
@@ -0,0 +1,24 @@
+use http::header::{HeaderMap, HeaderName, HeaderValue};
+
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq)]
+pub struct ModifyRequestHeader {
```

Comment:
nit/TIOLI: it seems like this type doesn't _really_ need to be named "modify *request* header", since there isn't actually anything request-specific about it? the `apply` method just mutates a `HeaderMap`, so it seems like the same struct could eventually be used to modify response headers too...it's the `Filter` enum that's going to be responsible for determining which `HeaderMap` is modified...